### ISSUE TITLE
[BUG]dask_cudf: read_csv seems to compute only first partition

### ISSUE THREAD

**Issue by MordicusEtCubitus** [rapidsai/cudf#9719]
Dears, 
Thanks for creating cudf and all dask work around that make it so great !

I'm trying to read a 8Go CSV file and doing a few computation using dask cudf and LocalCudaCluster
But I'm getting strange results, it seems that only first partition is used for computation
The file I'm reading can be downloaded from here:
http://data.cquest.org/geo_sirene/last/etablissements_actifs.csv.gz

Then, here is my code: the goal is to find pizza near my location in France - a funny exercice

```python

from dask_cuda import LocalCUDACluster  # I've got 2 GTX 1080ti GPU
from dask.distributed import Client, progress

cluster = LocalCUDACluster()
client = Client(cluster)
import dask_cudf as dc

ent = dc.read_csv("data/etablissements_actifs.csv", sep=","
                  , dtype={"l1_normalisee" : str, "geo_adresse" : str, "latitude" : float, "longitude" : float}
                  , usecols=["l1_normalisee", "latitude", "longitude", "geo_adresse"])

import geocoder
pos = geocoder.osm("Grande arche de la défense")

pizz = ent.dropna()
pizz = pizz[ pizz.l1_normalisee.str.contains("PIZZ") ]
pizz['distance'] = ((pizz.latitude-pos.lat)**2+(pizz.longitude-pos.lng)**2)**0.5
top10 = pizz.nsmallest(10, "distance")

r = top10.compute()
r
```

The result is returning shops that are not located in Paris at all but in departement 02, not really the same place :(

If I'm using the same file with dask.dataframe I got right results

It seems that only first partition is used, but not sure.
Below are my versions of librairies:

```python
print(dask_cudf.__version__)
print(dask.__version__)
print(dask_cuda.__version__)

21.10.01
2021.09.1
0+unknown  # strange
```

**Comment by MordicusEtCubitus** [comment]
Indeed, when playing with partitions, it appears that after partition 1, all records have an issue in columns content:

```python
ent.partitions.fn(1).head()
```
<div class="cell code_cell rendered selected" tabindex="2"><div class="output_wrapper"><div class="output"><div class="output_area"><div class="output_subarea output_html rendered_html output_result" dir="auto"><div>

l1_normalisee | latitude | longitude | geo_adresse
-- | -- | -- | --
423034149 | 25.0 | <NA> | <NA>
524459773 | 15.0 | <NA> | LO CRICCHIO FRANCK ESPACE VERT
391360112 | 18.0 | <NA> | LA CONDAMINE VILLAGE
398483081 | 12.0 | <NA> | CHEZ MR ABADIANE BT 6
880173638 | 11.0 | <NA> | LES MURIERS BAT C9


</div></div></div></div></div></div><div class="cell code_cell unrendered unselected" tabindex="2"><div class="input"><div class="prompt_container"></div></div></div><div class="prompt_container"></div><div class="inner_cell"><div class="input_area" aria-label="Edit code here"><div class="CodeMirror cm-s-ipython"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 5.59998px; left: 4px;"><textarea style="position: absolute; bottom: -1em; padding: 0px; width: 1px; height: 1em; outline: currentcolor none medium;" autocorrect="off" autocapitalize="none" spellcheck="false" tabindex="0" wrap="off"></textarea></div></div></div></div>

**Comment by beckernick** [comment]
Do you see similar behavior if you read the file with `cudf.read_csv`? Or only if you use `dask_cudf.read_csv`?

**Comment by MordicusEtCubitus** [comment]
I cannot check with cudf only as my GPU has 8Go of ram only and data does not fit into this memory 

But with an older version of the file the result is good, so it may be a file issue ; but this file provides good results with dask.dataframe

So I still in doubt.

**Comment by MordicusEtCubitus** [comment]
I've rent a GPU with enought memory and yes, it works fine with the same file and using directly cudf.read_csv in place of dask_cudf.read_csv

So apparently it is a dask_cudf issue.

Thanks in advance for helping

**Comment by rjzamora** [comment]
I will try to reproduce this soon, but I suspect the problem may be related to a `usecols` but that was already fixed in #9618

**Comment by rjzamora** [comment]
> I will try to reproduce this soon, but I suspect the problem may be related to a usecols bug that was already fixed in #9618

I had a chance to investigate and the problem is indeed the `usecols` bug that was fixed in #9618. Unfortunately, that fix does **not** resolve the issue when `dtype` is also specified. Therefore, we will need further changes before this issue can be closed - I plan to work on a solution today or tomorrow.

**Comment by MordicusEtCubitus** [comment]
Thanks a lot.  I've looked at commit for #9618 but do not understood why this is the issue, so not sure I can help writing the code to solve it

**Comment by rjzamora** [comment]
> so not sure I can help writing the code to solve it

No worries! I expect that #9796 will resolve your error - However, I'll be happy to iterate if this is not the case.

**Comment by rjzamora** [comment]
I closed this issue with #9796, but feel free to reopen if the problem persists.
